compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp

```diff
@@ -137,6 +137,97 @@ void asym_quant_bias(CircleConst *node, float input_scale, float weight_scale,
   *zp = 0;
 }
 
+bool get_channel_dim_index(CircleConst *node, loco::TensorShape &dimension, int &channel_dim_index)
+{
+  auto succs = loco::succs(node);
+  if (succs.size() != 1) // assume weights is used by only one node
+    return false;
+
+  for (auto out : succs)
+  {
+    auto conv = dynamic_cast<CircleConv2D *>(out);
+    auto dw_conv = dynamic_cast<CircleDepthwiseConv2D *>(out);
+    auto tw_conv = dynamic_cast<CircleTransposeConv *>(out);
+    auto fc = dynamic_cast<CircleFullyConnected *>(out);
+
+    if (conv != nullptr && conv->filter() == node)
+    {
+      assert(node->rank() == 4);
+      dimension.dim(0).set(node->dim(0).value());
+      dimension.dim(1).set(node->dim(1).value());
+      dimension.dim(2).set(node->dim(2).value());
+      dimension.dim(3).set(node->dim(3).value());
+      channel_dim_index = 3;
+      return true;
+    }
+    else if ((tw_conv != nullptr && tw_conv->filter() == node) ||
+             (dw_conv != nullptr && dw_conv->filter() == node))
+    {
+      assert(node->rank() == 4);
+      dimension.dim(0).set(node->dim(0).value());
+      dimension.dim(1).set(node->dim(1).value());
+      dimension.dim(2).set(node->dim(2).value());
+      dimension.dim(3).set(node->dim(3).value());
+      channel_dim_index = 2;
+      return true;
+    }
+    else if (fc != nullptr && fc->weights() == node)
+    {
+      assert(node->rank() == 2);
+      dimension.dim(0).set(1);
+      dimension.dim(1).set(1);
+      dimension.dim(2).set(node->dim(0).value());
+      dimension.dim(3).set(node->dim(1).value());
+      channel_dim_index = 3;
+      return true;
+    }
+    else
+    {
+      // node does not support channle-wise quantization
+      assert(false);
+    }
+  }
+
+  return false;
+}
+
+uint32_t cal_offset(loco::TensorShape &dimension, uint32_t *indices)
+{
+  return indices[0] * dimension.dim(1).value() * dimension.dim(2).value() *
+             dimension.dim(3).value() +
+         indices[1] * dimension.dim(2).value() * dimension.dim(3).value() +
+         indices[2] * dimension.dim(3).value() + indices[3];
+}
+
+void asym_quant_bias_per_channel(CircleConst *node, float input_scale,
+                                 std::vector<float> &weight_scale,
+                                 std::vector<float> &scaling_factor, std::vector<int64_t> &zp)
+{
+  float scale{0};
+
+  uint32_t size = node->size<loco::DataType::FLOAT32>();
+  std::vector<int32_t> quantized_values(size);
+
+  for (uint32_t i = 0; i < size; ++i)
+  {
+    scale = input_scale * weight_scale[i];
+    scaling_factor[i] = (scale == 0) ? 0 : 1.0 / scale;
```

Comment:
This looks like the inverse of scaling factor.

Reply:
`asym_quant_bias_per_layer` saves non-inversed scale as is.

Reply:
In case of ```asym_quant_bias_per_layer```, inversed scale is also used as [here](https://github.com/Samsung/ONE/blob/c6d68fb491d21bdc669024036f8455e378265e16/compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp#L116-L125)

Reply:
> This looks like the inverse of scaling factor. Is it intended?

For 8bit quantization, we used asymmetric quantization method.
So, 
we use equation as below
```
Out = Sin*Sw/Sout ((Input - Zin) * (Weight - Zw) + Bias/Sw*Sin)
```

Reply:
> In case of asym_quant_bias_per_layer, inversed scale is also used as here

It is used, but the saved scale is the non-inversed one.
https://github.com/Samsung/ONE/blob/c6d68fb491d21bdc669024036f8455e378265e16/compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp#L136

Reply:
oh, I see. I didn't catch up.